medicine_search_bot.py

```diff
@@ -0,0 +1,1269 @@
+                    one_time_keyboard=True,
+                    resize_keyboard=True,
+                    input_field_placeholder='Продовжуйте',
+                ),
+                caption='Ось відсканований штрихкод ✅:\n' + '<b>' + barcode + '</b>' + "\n\n"
+                        + format_query(query_result),
+            )
+
+        elif query_result:
+            update.message.reply_text(
+                parse_mode='HTML',
+                reply_markup=ReplyKeyboardMarkup(
+                    reply_keyboard,
+                    one_time_keyboard=True,
+                    resize_keyboard=True,
+                    input_field_placeholder='Продовжуйте',
+                ),
+                text='Ось відсканований штрихкод ✅:\n' + '<b>' + barcode + '</b>' + "\n\n"
+                     + format_query(query_result) + "\n\n⚠️ Фото відсутнє",
+                quote=True,
+            )
+        else:
+            reply_keyboard = [['Завершити сканування']]
+
+            update.message.reply_text(
+                parse_mode='HTML',
+                reply_markup=ReplyKeyboardMarkup(
+                    reply_keyboard,
+                    one_time_keyboard=True,
+                    resize_keyboard=True,
+                    input_field_placeholder='Продовжуйте',
+                ),
+                text='Штрих-код ' + '<b>' + barcode + '</b>'
+                     + ' на жаль відсутній у моїй базі даних ❌'
+                       '\n\nЯкщо ви хочете долучитись до наповнення бази даних - '
+                       'скористайтесь нашим другим ботом <b>@msb_database_bot</b>',
+                quote=True,
+            )
+
+        if context.user_data.setdefault("GOOGLE_SEARCH", "True") == "True":
+            update.message.reply_text(
+                parse_mode='HTML',
+                reply_markup=ReplyKeyboardMarkup(
+                    reply_keyboard,
+                    one_time_keyboard=True,
+                    resize_keyboard=True,
+                    input_field_placeholder='Продовжуйте',
+                ),
+                text='<b>' + '\n\nЙмовірно це: ' + '</b>' + get_query_heading(barcode)
+                     + ' - (За результатами пошуку ' + f'<a href="{link}"><b>Google</b></a>' + ')',
+                disable_web_page_preview=True,
+            )
+
+        context.user_data["DRUG_CODE"] = barcode
+
+
+def get_query_heading(barcode) -> str:
+    """
+    The get_query_heading function takes a string of the form '0123456789' and returns
+    the heading of the first Google search result for that string.
+
+    :param barcode: Barcode for the Google search
+    :return: A string containing the first heading of a search query
+    """
+    url = 'https://google.com/search?hl=uk&q=' + barcode
+
+    request_result = requests.get(url)
+    soup = bs4.BeautifulSoup(request_result.text, "html.parser")
+
+    heading_objects = soup.find_all('h3')
+
+    first_3_headings = heading_objects[0:3]
```

Comment:
Take a look at the [```limit```](https://www.crummy.com/software/BeautifulSoup/bs4/doc/#the-limit-argument) argument of ```BeautifulSoup.find_all``` method.
To optimize your code, consider using the following snippet:
```suggestion
    first_3_headings = soup.find_all('h3', limit = 3)
```

Reply:
Accepted